src/components/algolia-search/components/highlight.tsx

```diff
@@ -0,0 +1,42 @@
+import { parseAlgoliaHitHighlight } from '@algolia/autocomplete-preset-algolia'
+
+type HighlightHitProps<THit> = {
+  /**
+   * The Algolia hit whose attribute to retrieve the highlighted parts from.
+   */
+  hit: THit
+  /**
+   * The attribute to retrieve the highlighted parts from.
+   *
+   * You can use the array syntax to reference nested attributes.
+   */
+  attribute: keyof THit | string[]
```

Comment:
Yeah, it seems to be a semi-common convention to prefix generic type arguments with `T`. I'm open to any suggestions if you feel it could be more clear!

Reply:
Okay cool! I just hadn't seen it before, but it makes sense 👍